web/modules/custom/server_general/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Drupal\server_general\Plugin\EntityViewBuilder;
+
+use Drupal\intl_date\IntlDate;
+use Drupal\media\MediaInterface;
+use Drupal\node\Entity\NodeType;
+use Drupal\node\NodeInterface;
+use Drupal\server_general\EntityDateTrait;
+use Drupal\server_general\EntityViewBuilder\NodeViewBuilderAbstract;
+use Drupal\server_general\LineSeparatorTrait;
+use Drupal\server_general\SocialShareTrait;
+use Drupal\server_general\TitleAndLabelsTrait;
+use Drupal\Core\Link;
+use \Drupal\Core\Url;
+use Drupal\og\OgMembershipInterface;
+
+
+/**
+ * The "Node Group" plugin.
+ *
+ * @EntityViewBuilder(
+ *   id = "node.group",
+ *   label = @Translation("Node - Group"),
+ *   description = "Node view builder for Group bundle."
+ * )
+ */
+class NodeGroup extends NodeViewBuilderAbstract {
+
+  use EntityDateTrait;
+  use LineSeparatorTrait;
+  use SocialShareTrait;
+  use TitleAndLabelsTrait;
+
+  /**
+   * Build full view mode.
+   *
+   * @param array $build
+   *   The existing build.
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array.
+   */
+  public function buildFull(array $build, NodeInterface $entity) {
+    $elements = [];
+
+    // Header.
+    $element = $this->buildHeader($entity);
+    $elements[] = $this->wrapContainerWide($element);
+
+    // Main content and sidebar.
+    $element = $this->buildMainAndSidebar($entity);
+    $elements[] = $this->wrapContainerWide($element);
+
+    $elements = $this->wrapContainerVerticalSpacingBig($elements);
+    $build[] = $this->wrapContainerBottomPadding($elements);
+
+    return $build;
+  }
+
+  /**
+   * Build the header.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array
+   *
+   * @throws \IntlException
+   */
+  protected function buildHeader(NodeInterface $entity): array {
+    $elements = [];
+
+    $elements[] = $this->buildConditionalPageTitle($entity);
+
+    // Show the node type as a label.
+    $node_type = NodeType::load($entity->bundle());
+    $elements[] = $this->buildLabelsFromText([$node_type->label()]);
+
+    $current_user = \Drupal::currentUser();
+    //$logged_in = \Drupal::currentUser()->isAuthenticated();
+
+    if ($current_user->isAuthenticated()) {
+      $account = \Drupal\user\Entity\User::load($current_user->id());; // pass your uid
+      $name = $account->get('name')->value;
+
+      $parameters = [
+          'entity_type_id' => $entity->getEntityTypeId(),
+          'group' => $entity->id(),
+          'og_membership_type' => OgMembershipInterface::TYPE_DEFAULT,
+        ];
+
+      $url = Url::fromRoute('og.subscribe', $parameters);
+
+      $link['title'] = $this->t("Hi @user, click here if you would like to subscribe to this group called @groupname", ['@user'=> $name, '@groupname' => $entity->label()]);
```

Comment:
Where's the link to subscribe? the "click here" part....

Reply:
I have set link on entire texts using below code :
$elements[] = [
        '#type' => 'link',
        '#title' => $link['title'],
        '#url' => $link['url'],
      ];

Reply:
:+1: 

Let's have that one tested.

Reply:
I have fixed link using markup. Please check last commit